pkg/logging/types.go

```diff
@@ -0,0 +1,10 @@
+package logging
+
+type Logger interface {
+	Debug(msg string)
+	Info(msg string)
+	Warn(msg string)
+	Error(msg string)
```

Comment:
```suggestion
	Debug(args ...interface{})
	Info(args ...interface{})
	Warn(args ...interface{})
	Error(args ...interface{})
```

We should probably allow for other types besides string. If I remember correctly, the loggers follows the signature of fmt's `Print()` function. As in `Print(a ...interface{}`.

In the logrus wrapper, it would look like this:
![image](https://user-images.githubusercontent.com/1145370/144593461-623fba2b-58bf-4c2b-ae60-cc7c280b888a.png)

Reply:
Yes, but if we switch to e.g. `Zap` logging library it only accepts a single message (string) + a set of key-value pairs for structured logging, which we support with `WithField`.

What do you want multiple arguments for here? To avoid having to format your message yourself?

Reply:
You can always use `fmt.Sprint` or `Sprintf` for that too.

Reply:
Never mind, I see that `Zap` also supports `Info(args ...interface{})` in their `SugaredLogger`.

Reply:
But, by restricting the API here, we "force" people to put the values in fields instead, which gives us the structured logging we want?

Reply:
Will it be structured unless we force the keys of the fields as well?

Anywho. Valid points all around. I'd use the ...interface{} args version mostly due to having seen it in multiple established places ([go logging lib](https://github.com/golang/go/blob/master/src/log/log.go#L210), logrus, zap). String seems fine enough though. Thoughts @yngvark ?

Reply:
The keys of the fields in `WithField` are strings, if that's what you mean @deifyed? Or do you mean that we standardize names of keys (which could be a good idea anyway)?

Reply:
Last part, standardizing names of keys. I'd argue against standardizing them at this moment since we don't have need for structured logging right now. It was just an argument invalidating the structured part. String is fine for now IMO. 

A use case for using args interface{} is that we could bake in marshalling(serializing) so that we wouldn't have to do this outside the log function everytime we'd want to log a struct for example

Reply:
> Anywho. Valid points all around. I'd use the ...interface{} args version mostly due to having seen it in multiple established places (go logging lib, logrus, zap).

I agree with this. We're trying to create an interface that abstracts common logging libraries. By not deviating too far from what other loggers do, we're more resilient to future needs for our logging.

Perhaps `WithFields` covers our need for structured logging, but there might be other needs that it doesn't solve. Maybe we want to use a custom logger that always takes some struct as input, for instance.

I don't see any reason to use `string` instead of `interface{}`, other than maybe giving our API clearer intention of use.

On the other hand, we could always change from `string` to `interface{}` in the future without much hazzle, it's the other way around that hurts.